lib/middleware/request_tracker.rb

```diff
@@ -61,7 +61,7 @@ def self.log_request(data)
       elsif data[:has_auth_cookie]
         ApplicationRequest.increment!(:page_view_logged_in)
         ApplicationRequest.increment!(:page_view_logged_in_mobile) if data[:is_mobile]
-      else
+      elsif !SiteSetting.login_required || !SiteSetting.suppress_anonymous_pageviews_on_private_sites
```

Comment:
This will exclude anonymous views for login required sites regardless of the setting, right? (Or I am reading this wrong? I have trouble parsing "not this OR not that" in my head.

Reply:
> I have trouble parsing "not this OR not that" in my head.

Double negations is super hard 🤯 

> This will exclude anonymous views for login required sites regardless of the setting, right?

Looks like it.

Reply:
Yes, exactly. At first I wanted to call it just `suppress_anonymous_pageviews`, but I wanted to point out the site has to be private (login_required = true).

Reply:
It’s fine we can include that in the description of the setting 😉

Reply:
Hmm... it's a hidden setting though.

Reply:
That’s a very good point. Still I much prefer the shorter name. If they want to change it, they will need to do it via the console/ENV, so they will have to understand what it does